server_src/runners/history.js

```diff
@@ -572,8 +541,8 @@ class HistoryCollection {
         name = printer.printerURL;
       }
       if (
-        serverSettings[0].filamentManager &&
-        Array.isArray(printer.selectedFilament)
+        serverSettings[0]?.filamentManager &&
```

Comment:
This is checked multiple times. Also I thought serverSettings are cached? Shouldnt we pull from cache instead of database?

Reply:
Yes but saving for the history re-write. wanted this as a quick fix.

Reply:
Well settings re-write should I say mentioned in another comment